### PR TITLE
New: added successful reset event trigger (fixes #335)

### LINKED ISSUE #335: AdaptModel reset trigger
Currently it isn't easy to know when an incomplete model with children is reset. We can listen to a `change:_isInteractionComplete` or  `change:_isComplete` to know when the status has been changed, but this won't inform us of reset being called when the parent status is unchanged.

For plugins which modify child models of an associated parent, or work with a collection of models like an assessment, it would be useful to know when a reset has been called on that parent model regardless of whether the completion status changed.

### Proposed solution
Add `this.trigger('reset', this);` following a successful reset.

### PR DESCRIPTION
**Pull request by danielghost**
Fixes #335

### New
* Added a successful `'reset'` event trigger for AdaptModel

### CODE REVIEW

File: js/models/adaptModel.js [hunk 1]
```diff
@@ -228,23 +228,17 @@ export default class AdaptModel extends LockingModel {
...
-        return true;
-    }
-    return false;
+    const resetData = {
+      _isEnabled: true,
+      _isInteractionComplete: false
+    };
+    const isHardReset = (type === 'hard' || type === true);
+    const isSoftReset = (type === 'soft');
+    if (!isHardReset && !isSoftReset) return false
```
Comment (cahirodoherty-learningpool) [inline js/models/adaptModel.js:237]: Minor: You can move the resetData object to after this early return. It isn't used otherwise

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by joe-allen-89 — APPROVED** [review]
Tested with latest FW and core, working as expected.

**Pull request merged**